lib/idm-ts.ts

```diff
@@ -60,23 +80,29 @@ export class IDMObject<T extends IDMObjectType<string>, D extends IDMObjectType<
   public patch<F extends Fields<T>>(
     id: string,
     rev: string | null,
-    value: PatchOpts[],
+    value: CompositePatchOpts<T>,
     options: { readonly params?: object; readonly fields: F[] }
   ): ResultType<T, F>;
   public patch<F extends Fields<T>>(
     id: string,
     rev: string | null,
-    value: PatchOpts[],
+    value: CompositePatchOpts<T>,
     options: { readonly params?: object; readonly unCheckedFields: F[] }
   ): T & Revision;
-  public patch<F extends Fields<T>>(id: string, rev: string | null, value: PatchOpts[], options?: { readonly params?: object }): D & Revision;
+  public patch<F extends Fields<T>>(id: string, rev: string | null, value: CompositePatchOpts<T>, options?: { readonly params?: object }): D & Revision;
   public patch<F extends Fields<T>>(
     id: string,
     rev: string | null,
-    value: PatchOpts[],
+    value: CompositePatchOpts<T>,
     { params, fields, unCheckedFields }: { readonly params?: object; readonly fields?: F[]; readonly unCheckedFields?: string[] } = {}
   ) {
-    return openidm.patch(`${this.type}/${id}`, rev, value, params, unCheckedFields ? unCheckedFields : fields);
+    let patchValues: PatchOpts[];
```

Comment:
nit: make this a ternary so you don't have the `let`:

```typescript
const patchValues = this.isCombinedPatchOpts(value)
  ? [...value.unCheckedPatches, ...value.checkedPatches ?? []]
  : value;
```

Reply:
I had considered that actually, and couldn't decide if it was too long. But I guess with Agile being for immutability I should have gone that way ;)